### PR TITLE
add to mcc detect and infer test

### PR DESCRIPTION
**Pull request by AleksandrPanov**
merge with https://github.com/opencv/opencv_extra/pull/1153

Added a full pipeline tests:

1. detector->process(img, (TYPECHART)0, 1, true);
2. ColorCorrectionModel model(src, COLORCHECKER_Macbeth); model.run();
3. calibratedImage = model.infer(calibratedImage)*255.;

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/mcc/test/test_mcc.cpp [hunk 1]
```diff
@@ -81,5 +81,54 @@ TEST(CV_mccRunCCheckerDetectorBasic, accuracy_VINYL18)
     runCCheckerDetectorBasic("VINYL18.png", VINYL18);
 }
 
+TEST(CV_mcc_ccm_test, detectAndInfer)
+{
+    string path = cvtest::findDataFile("mcc/mcc_ccm_test.jpg");
+    Mat img = imread(path, IMREAD_COLOR);
+    Ptr<CCheckerDetector> detector = CCheckerDetector::create();
```
[the remark this answers is not in the document]
Comment (AleksandrPanov) [reply modules/mcc/test/test_mcc.cpp:88]: `findDataFile` will break the test if it does not find the data

File: modules/mcc/test/test_mcc.cpp [hunk 2]
```diff
@@ -81,5 +81,54 @@ TEST(CV_mccRunCCheckerDetectorBasic, accuracy_VINYL18)
...
+    ASSERT_MAT_NEAR(gold_ccm, ccm, 2e-3);
+
+    const double gold_loss = 4.6386569120323129;
+    const double loss = model.getLoss();
+    // check loss
+    EXPECT_NEAR(gold_loss, loss, 1e-2);
+
+    // read gold calibrate img
+    path = cvtest::findDataFile("mcc/mcc_ccm_test_res.png");
+    Mat gold_img = imread(path);
```
[the remark this answers is not in the document]
Comment (AleksandrPanov) [reply modules/mcc/test/test_mcc.cpp:154]: `findDataFile` will break the test if it does not find the data

**Review by asmorkalov — APPROVED** [review]
:+1:

**Review by asmorkalov — APPROVED** [review]
:+1:
